advanced/wallets/react-wallet-v2/src/hooks/useSmartAccounts.ts

```diff
@@ -19,19 +20,52 @@ export default function useSmartAccounts() {
 
   const initializeSmartAccounts = async (privateKey: string) => {
     if (smartAccountEnabled) {
-      if (kernelSmartAccountEnabled) {
-        const { kernelSmartAccountAddress } = await createOrRestoreKernelSmartAccount(privateKey)
-        SettingsStore.setKernelSmartAccountAddress(kernelSmartAccountAddress)
-      }
-      if (safeSmartAccountEnabled) {
-        const { safeSmartAccountAddress } = await createOrRestoreSafeSmartAccount(privateKey)
-        SettingsStore.setSafeSmartAccountAddress(safeSmartAccountAddress)
-      }
-      if (biconomySmartAccountEnabled) {
-        const { biconomySmartAccountAddress } = await createOrRestoreBiconomySmartAccount(
-          privateKey
-        )
-        SettingsStore.setBiconomySmartAccountAddress(biconomySmartAccountAddress)
+      try {
+        const promises = [];
+        
+        if (kernelSmartAccountEnabled) {
```

Comment:
whats the difference between `smartAccountEnabled` and `kernelSmartAccountEnabled`